Shopify/cli
# Add a rate-limiter dependency for partners requests

### WHY are these changes introduced?
We need a way to limit requests to the API so that we don't hit the rate limit.
Partners API is limited to 10 requests per second and there are some special use cases that might launch more than that.



### WHAT is this pull request doing?
- Add a bottleneck feature to partners API, only 1 request is allowed every 100ms, if say 5 requests are launched at the same time, the bottleneck will process 1 every 100ms.


### How to test your changes?
- Create an app with 30-35 functions and try to deploy with `main` and with this branch. With main you should get a 429 error. With this branch it should work.


### Post-release steps
Reduce API rate-limit in partners to 10.



### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
